tls/extensions/s2n_extension_list.h

```diff
@@ -21,8 +21,8 @@
 #define S2N_PARSED_EXTENSIONS_COUNT S2N_SUPPORTED_EXTENSIONS_COUNT
 
 typedef struct {
-    uint16_t extension_type;
     struct s2n_blob extension;
+    uint16_t extension_type;
```

Comment:
Adding a new extension adds a new element to the client_hello parsed extension list, which adds ~40 bytes to the connection. That triggered the s2n_connection_size_test to yell at me for increasing the size of the connection. To avoid that, I swapped these two fields.

In C, the fields of a struct have to meet alignment requirements. If they don't, the compiler inserts empty space so that they do. By swapping extension and extension_type, I fixed the field alignment and avoided that empty space. See http://www.catb.org/esr/structure-packing/

Basically, this is an easy memory win that reduces the size of the parsed extensions array enough to offset the new element I added to it.

Reply:
See here: https://godbolt.org/z/f9EE56rP3